### PR TITLE
[RateLimiter] Adding annotations

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 5.x 
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | no
| License       | MIT
| Doc PR        | 

Following the example of the `SlidingWindow`'s PR reviews, I would like to add annotations for all properties. This makes it simpler to read the code and understand what is what.

### CODE REVIEW

File: src/Symfony/Component/RateLimiter/FixedWindowLimiter.php [hunk 1]
```diff
@@ -24,10 +24,29 @@
...
+     */
     private $id;
+
+    /**
+     * @var int
+     */
     private $limit;
+
+    /**
+     * @var \DateInterval
```
Comment (derrabus) [inline src/Symfony/Component/RateLimiter/FixedWindowLimiter.php]:
```suggestion
     * @var int
```

Comment (Nyholm) [reply src/Symfony/Component/RateLimiter/FixedWindowLimiter.php]: Yes, You are correct. I missed that we converted the DateInterval to an int in the constructor. 

Is this proof that annotations is really useful? =)

File: src/Symfony/Component/RateLimiter/Rate.php [hunk 2]
```diff
@@ -22,7 +22,14 @@
  */
 final class Rate
 {
+    /**
+     * @var \DateInterval
```
Comment (nicolas-grekas) [inline src/Symfony/Component/RateLimiter/Rate.php]: When a property matches a constructor argument, we prefer removing these annotations.
The IDE understands the type correctly, and there is less room for typos. The less lines in a code base, the less PRs on it, the less maintenance for us.

Comment (Nyholm) [reply src/Symfony/Component/RateLimiter/Rate.php]: I'll remove some of these then. 

Thank you

**Comment by Nyholm** [comment]
I removed the annotations that are the same as the constructor. On a few occasions i kept the annotations because the doc block had a valuable comment too.

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/RateLimiter/FixedWindowLimiter.php [hunk 3]
```diff
@@ -26,8 +26,16 @@ final class FixedWindowLimiter implements LimiterInterface
 {
     private $id;
     private $limit;
-    private $interval;
     private $storage;
+
+    /**
+     * @var int seconds
```
Comment (chalasr) [inline src/Symfony/Component/RateLimiter/FixedWindowLimiter.php:32]: When the description is at the property level, IDE (PHPStorm at least) does not use it to document the related constructor arg when the the constructor signature is displayed for help.  Considering this, would it make sense to rather add it as an `@param` annotation to the constructor? I think it'd be more consistent with what already have in the codebase.

Comment (Nyholm) [reply src/Symfony/Component/RateLimiter/FixedWindowLimiter.php:32]: This makes sense. 

The constructor takes a `\DateInterval`. That is why I added `int` on this property to be more clear what it contains.

Comment (chalasr) [reply src/Symfony/Component/RateLimiter/FixedWindowLimiter.php:32]: 👍  Missed it, thanks

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @Nyholm.

**Pull request merged**